controller/volume_controller.go

```diff
@@ -940,6 +942,16 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, e *longhorn
 			}
 		}
 
+		if v.Status.RemountRequired {
+			if err := util.RemountVolume(v.Name); err != nil {
+				return err
+			}
+			v.Status.RemountRequired = false
```

Comment:
Make sure when user detached the volume the RemountRequired field would be reset to false.